### PR TITLE
New User Authentication 

### PR DESCRIPTION
**Pull request by LamboCreeper**
This implements the feature of authenticating new users. This _should_ help us filter out suspicious users. We have to cache the authentication message so that the bot knows it exists on load.

Having this part open-source may cause issues where people can see how to break the authentication (not like it's hard to work out from within Discord), so eventually, we may do a GET request a private API which decides if a user is valid based on more detailed logic.

Big thanks to @jacobgasser for helping me with the message caching for detecting the reaction.

~_I kind of messed up this pull request so it contains code from a previous pull request (#21), please ignore that._~

### CODE REVIEW

File: events/guildMemberAdd.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Dependencies
+const app = require("./../app.js");
+
+const config = app.config;
+
+// The code that runs when the event is executed.
+function run(member) {
+	if (member.user.avatar) {
```
Comment (jacobgasser) [inline events/guildMemberAdd.js]: You ever seen those fake cam whore accounts? They all have busty pictures.

**Review by jacobgasser — CHANGES_REQUESTED** [review]

File: events/guildMemberAdd.js (hunk 1 above)
Comment (LamboCreeper) [reply events/guildMemberAdd.js]: Will change it so that their account must also be 30 days old.

File: events/guildMemberAdd.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const config = app.config;
+
+// The code that runs when the event is executed.
+function run(member) {
+	const currentDate = Date.now();
+	const memberCreatedDate = member.user.createdAt.getTime();
+	const dateDifference = currentDate - memberCreatedDate;
+
+	// Check that the member has an avatar and their account is at least 30 days old.
+	if (member.user.avatar && dateDifference / 2592000000 > 1) {
```
Comment (jacobgasser) [inline events/guildMemberAdd.js:13]: LAMBO! I told you...what about sleeper bots. They could claim to be and by all accounts seem like a normal user and even have a profile picture.

**Review by jacobgasser — CHANGES_REQUESTED** [review]

**Review by jacobgasser — APPROVED** [review]

File: events/guildMemberAdd.js (hunk 2 above)
Comment (LamboCreeper) [reply events/guildMemberAdd.js:13]: As discussed in Discord this is a feature that can be implemented later and is not of importance right now.

**Pull request merged**